core/src/main/java/com/aem/demo/core/services/impl/RssFeedServiceImpl.java

```diff
@@ -169,4 +175,25 @@ private Set<String> getSavedNodeNames(String rootPath, PageManager pageManager)
         return names;
     }
 
+    private String[] makeTags(ResourceResolver resourceResolver, NewsCard card) {
+        String[] pathArr = new String[3];
+        List<String> titles = splitTitle(card.getTopic());
+        TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+        try {
+            for (int i = 0; i < titles.size(); i++) {
+                String title = titles.get(i);
+                Tag tag = tagManager.createTag(title, title, card.getTopic(), true);
+                pathArr[i] = tag.getPath();
+            }
+        } catch (InvalidTagFormatException e) {
+            log.error("Exception occurred while creating tags " + e.getMessage());
+        }
+        return pathArr;
+    }
+
+    private List<String> splitTitle(String title) {
+        String regex = "([,\\s?.'’@]+)|(the)|(for)|(with)|(from)|(by)|(gets)";
+        List<String> stringList = List.of(title.toLowerCase().replaceAll(regex, " ").split(" "));
```

Comment:
With a replacement like this, `Alf from the space` will be converted to `Alf  space` and split into `["Alf", "", "space"]` (that's because of two spaces in a row). You need to additionally filter out empty strings, then.